src/reducers/versions.tsx

```diff
@@ -549,37 +549,58 @@ export type GetRelativeDiffAnchorParams = {
 };
 
 export const getRelativeDiffAnchor = ({
-  currentAnchor,
+  currentAnchor = '',
   diff,
   position = RelativePathPosition.next,
 }: GetRelativeDiffAnchorParams): string | null => {
   const anchors = getDiffAnchors(diff);
   if (anchors.length) {
-    let newIndex;
     if (!currentAnchor) {
       // Since we aren't looking for an anchor relative to an existing one,
       // just get the first anchor.
-      newIndex = 0;
-    } else {
-      const currentIndex = anchors.indexOf(currentAnchor);
-      if (currentIndex < 0) {
-        throw new Error(
-          `Could not locate anchor: ${currentAnchor} in the diff.`,
-        );
-      }
+      return anchors[0];
+    }
 
-      newIndex =
+    const currentIndex = anchors.indexOf(currentAnchor);
+
+    if (currentIndex >= 0) {
+      const newIndex =
         position === RelativePathPosition.previous
           ? currentIndex - 1
           : currentIndex + 1;
+
+      if (newIndex >= 0 && newIndex < anchors.length) {
+        return anchors[newIndex];
+      }
+      // The currentAnchor is the only anchor in the diff.
+      return null;
     }
 
-    if (newIndex >= 0 && newIndex < anchors.length) {
-      return anchors[newIndex];
+    // We have a currentAnchor, but is doesn't match anything in anchors, so
+    // we need to try to find the correct anchor closest to the currentAnchor.
+    const currentAnchorNumber = extractNumber(currentAnchor);
```

Comment:
Hmm, this seems like it wouldn't work in some edge cases where the normal line number might be higher than the inserted line number. I'd think it would be more robust to make a line-based map of all changes (maybe with [`ForwardComparisonMap`](https://github.com/mozilla/addons-code-manager/blob/d2aa422453107fde413c5a34ec741d5b699bb68d/src/pages/Compare/utils.tsx#L27)) and use a pointer to jump around within that map.

However, I can't find any real world diff that will demonstrate that kind of problem 🤷‍♀ 

Ship it. If we run into a real world problem we could revisit this logic.